pandas_gbq/gbq.py

```diff
@@ -188,78 +172,103 @@ def get_application_default_credentials(self):
         """
-        import httplib2
-        try:
-            from googleapiclient.discovery import build
-        except ImportError:
-            from apiclient.discovery import build
-        try:
-            from oauth2client.client import GoogleCredentials
-        except ImportError:
-            return None
+        credentials, _ = google.auth.default(scopes=[self.scope])
+        return self.try_credentials(credentials)
 
-        try:
-            credentials = GoogleCredentials.get_application_default()
-        except:
-            return None
+    def load_user_account_credentials(self):
+        """
+        Loads user account credentials from a local file.
 
-        http = httplib2.Http()
-        try:
-            http = credentials.authorize(http)
-            bigquery_service = build('bigquery', 'v2', http=http)
-            # Check if the application has rights to the BigQuery project
-            jobs = bigquery_service.jobs()
-            job_data = {'configuration': {'query': {'query': 'SELECT 1'}}}
-            jobs.insert(projectId=self.project_id, body=job_data).execute()
-            return credentials
-        except:
-            return None
+        Parameters
+        ----------
+        None
 
-    def get_user_account_credentials(self):
-        from oauth2client.client import OAuth2WebServerFlow
-        from oauth2client.file import Storage
-        from oauth2client.tools import run_flow, argparser
+        Returns
+        -------
+        - GoogleCredentials,
+            If the credentials can loaded. The retrieved credentials should
+            also have access to the project (self.project_id) on BigQuery.
+        - OR None,
+            If credentials can not be loaded from a file. Or, the retrieved
+            credentials do not have access to the project (self.project_id)
+            on BigQuery.
+        """
+        with open('bigquery_credentials.dat') as credentials_file:
+            credentials_json = json.load(credentials_file)
+            credentials = google.oauth2.credentials.Credentials(
+                token=credentials_json.get('access_token'),
+                refresh_token=credentials_json.get('refresh_token'),
+                id_token=credentials_json.get('id_token'),
+                token_uri=credentials_json.get('token_uri'),
+                client_id=credentials_json.get('client_id'),
+                client_secret=credentials_json.get('client_secret'),
+                scopes=credentials_json.get('scopes'))
+
+            # Refresh the token before trying to use it.
+            http = httplib2.Http()
+            request = google_auth_httplib2.Request(http)
+            credentials.refresh(request)
+
+        return self.try_credentials(credentials)
+
+    def save_user_account_credentials(self, credentials):
+        """
+        Saves user account credentials to a local file.
+        """
+        with open('bigquery_credentials.dat', 'wb') as credentials_file:
```

Comment:
This method will now log a message if the file cannot be saved. Since this doesn't prevent the credentials being used, it no longer throws an exception.

I've added a test for the case.